vsphere/vsphere.go

```diff
@@ -201,7 +204,12 @@ func (ops *vsphereOps) Attach(diskPath string) (string, error) {
 		return "", err
 	}
 
-	diskUUID, err := vmObj.AttachDisk(ctx, diskPath, &vclib.VolumeOptions{SCSIControllerType: vclib.PVSCSIControllerType})
+	volOpts := &vclib.VolumeOptions{SCSIControllerType: vclib.PVSCSIControllerType}
+	attachMode, ok := options[DiskAttachMode]
+	if ok {
+		volOpts.DiskType = attachMode
```

Comment:
@harsh-px vsphere lib changes for disk mode is in previous PR - >https://github.com/libopenstorage/cloudops/pull/59/files#diff-f5678f4d3303f39298329cfc45507b5eR308